source/Ship.h

```diff
@@ -198,6 +199,10 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	std::string GetHail(std::map<std::string, std::string> &&subs) const;
 	bool CanSendHail(const PlayerInfo &player, bool allowUntranslated = false) const;
 
+	// Access the ship's AI cache, containing the range and expected AI behavior for this ship.
+	ShipAICache &GetAICache();
+	void ReCacheInfo();
```

Comment:
How about this function name instead?
```suggestion
	void UpdateCaches();
```